components/script/dom/htmlinputelement.rs

```diff
@@ -886,6 +873,23 @@ impl HTMLInputElement {
             _ => ()
         }
     }
+
+    fn selection_direction(&self) -> SelectionDirection {
+        self.textinput.borrow().selection_direction
+    }
+
+    fn set_selection_range(&self, start: u32, end: u32, direction: SelectionDirection) {
```

Comment:
Let's add a spec link comment above this function (https://html.spec.whatwg.org/multipage/#set-the-selection-range).